simulator/Simulator.kt

```diff
@@ -119,8 +119,13 @@ open class Simulator(
         state.mem = mem
     }
 
-    fun run() {
+    fun run(plugins: List<IsSimulatorPlugin>) {
         while (!isDone()) {
+            if(!plugins.isEmpty()) {

```

Comment:
this code should exist in step and not run as this should happen per step.

Reply:
I can move it into step